e2e/alert_action.spec.ts

```diff
@@ -0,0 +1,66 @@
+import {test, expect, Page} from '@playwright/test';
+import {dataSetSearch} from './utils';
+import {setTimeout} from 'timers/promises';
+
+test('Alert action - create and delete alert with results propagation to DataSet', async ({page}) => {
+    const serverHost = 'host_splunk';
+    const alertName = 'test_alert';
+    await removeAlertIfExists(page, alertName);
+
+    await dataSetSearch(page, "| dataset");
+    await saveAsAlertWithDataSetTrigger(page, alertName, serverHost);
+    await removeAlertIfExists(page, alertName);
+});
+
+async function saveAsAlertWithDataSetTrigger(page: Page, alertName: string, serverHost: string) {
+    console.log("Open Save As Alert Dialog")
+    await page.getByText("Save As").click();
+    await page.getByRole('link', {name: 'Alert'}).click();
+
+    console.log("Fill in Alert Dialog - general");
+    await page.locator('[name="name"]').fill(alertName);
+    await page.getByRole('link', {name: 'Run every week'}).click();
+    await page.getByRole('link', {name: 'Run on Cron Schedule'}).click();
+    await page.locator('[name="cron_schedule"]').fill("*/1 * * * *"); // run every minute
+    // by default triggered if num of results > 0
+    await page.screenshot({path: `playwright-screenshots/page-alert-dialog-general.png`, fullPage: true});
+
+    console.log("Add trigger action - Dataset");
+    await page.getByText("+ Add Actions").click();
+    await page.getByText("Send to DataSet").click();
+    await page.screenshot({path: `playwright-screenshots/page-alert-dialog-dataset-trigger.png`, fullPage: true});
+    await expect(page.getByLabel(`ServerHost`)).toBeVisible();
+    await expect(page.getByLabel(`ServerHost`)).toBeVisible();
+    await expect(page.getByLabel(`DataSet Message`)).toBeVisible();
+    await expect(page.getByLabel(`Severity`)).toBeVisible();
+    await expect(page.getByLabel(`ServerHost`)).toBeVisible();
+
+    console.log("Fill in Alert Dialog - trigger DataSet event");
+    await setTimeout(500); // wait for select option to be populated
+    await page.locator('[name="action.dataset_event.param.account"]').locator('button').click();
+    await page.locator('[data-test="option"]').first().click(); // select first account
+    await page.locator('[name="action.dataset_event.param.dataset_serverhost"]').fill(serverHost)
```

Comment:
When we are filling the server host, we are imho implicitly checking, that it's there and is visible. At least this is my expectation, so those checks above are redundant?

Reply:
true, I would prefer to keep that previous assertion so its consistent (verification that all required elements are there all in one place)